pageserver/src/basebackup.rs

```diff
@@ -157,6 +167,58 @@ where
         Ok(())
     }
 
+    fn add_rel(&mut self, tag: RelTag) -> anyhow::Result<()> {
```

Comment:
Nit: I had some fun trying to simplify. Feel free to copy anything from [here](https://github.com/neondatabase/neon/compare/fullbackup...bojan/fullbackup?expand=1).